src/services/pagination/LengthAwarePaginator.js

```diff
@@ -0,0 +1,81 @@
+import Paginator from './Paginator';
+import LesgoException from '../../exceptions/LesgoException';
+
+const FILE = 'Services/pagination/LengthAwarePaginator';
+
+export default class LengthAwarePaginator extends Paginator {
+  /**
+   * Constructor
+   *
+   * @param db
+   * @param sql
+   * @param sqlParams
+   * @param perPage
+   * @param currentPage
+   * @param total
+   */
+  constructor(db, sql, sqlParams, perPage = 10, currentPage = 1, total = null) {
```

Comment:
I just noticed this. But I'm not sure if you guys have this rule. Most of the time we apply to max-params for any given class or function. If it's more than 3 I'll convert this to be an object instead. The reason being is that you might miss the order of params even tho some of them are optional.

Reply:
Yep. I actually just realized that a few days ago. I can refactor it though, if it is still possible at this point.

Reply:
My suggestion would be to have perPage, currentPage, total in the additional params object since these are optional